src/client/sandbox/fetch.ts

```diff
@@ -9,18 +9,15 @@ import SAME_ORIGIN_CHECK_FAILED_STATUS_CODE from '../../request-pipeline/xhr/sam
 import { overrideDescriptor } from '../utils/property-overriding';
 import * as browserUtils from '../utils/browser';
 import { transformHeaderNameToBuiltin, transformHeaderNameToInternal } from '../utils/headers';
+import CookieSandbox from './cookie';
 
-const DEFAULT_REQUEST_CREDENTIALS = nativeMethods.Request ? new nativeMethods.Request(window.location.toString()).credentials : void 0;
+const DEFAULT_REQUEST_CREDENTIALS = nativeMethods.Request ? new nativeMethods.Request(location.toString()).credentials : void 0;
```

Comment:
I propose to create the `getGlobalContextInfo` function and use it in all necessary places.

```js
export default getGlobalContextInfo (): GlobalContextInfo {
    const isInWorker = typeof window === 'undefined';

   return {
         isInWorker,
         global: isInWorker ? self : window
   };
}
```